pkg/controller/nodeipam/ipam/cidrset/cidr_set.go

```diff
@@ -88,14 +88,17 @@ func NewCIDRSet(clusterCIDR *net.IPNet, subNetMaskSize int) (*CidrSet, error) {
 	registerCidrsetMetrics()
 
 	maxCIDRs = 1 << uint32(subNetMaskSize-clusterMaskSize)
-	return &CidrSet{
+	cidrSet := &CidrSet{
 		clusterCIDR:     clusterCIDR,
 		nodeMask:        net.CIDRMask(subNetMaskSize, bits),
 		clusterMaskSize: clusterMaskSize,
 		maxCIDRs:        maxCIDRs,
 		nodeMaskSize:    subNetMaskSize,
 		label:           clusterCIDR.String(),
-	}, nil
+	}
+	cidrSetMax.WithLabelValues(cidrSet.label).Add(float64(maxCIDRs))
```

Comment:
@logicalhan and @dgrisonnet for advice on how to define a metrics that represent a constant ... I remember they linked me to some place in another review but I can't find the reference now :upside_down_face:

Reply:
you would usually want to create a const metric via [NewConstMetric](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#NewConstMetric)

Reply:
not using this, since a limit can increase or decrease as you explained in https://github.com/kubernetes/kubernetes/pull/112260#discussion_r966351874